lib/teiserver/battle/balance/split_one_chevs.ex

```diff
@@ -0,0 +1,205 @@
+      %{count: 1, members: [3], group_rating: 7, ratings: [7]}
+  ]
+
+  Output:  [
+  %{rating: 8, member_id: 1},
+  %{rating: 5, member_id: 4},
+  %{rating: 6, member_id: 2},
+  %{rating: 7, member_id: 3}
+  ]
+  """
+  def flatten_members(expanded_group) do
+    for %{members: members, ratings: ratings} <- expanded_group,
+        # Zipping will create binary tuples from 2 lists
+        {id, rating} <- Enum.zip(members, ratings),
+        # Create result value
+        rank = get_rank(id),
+        do: %{member_id: id, rating: rating, rank: rank}
+  end
+
+  def get_rank(member_id) do
+    Account.get_user_by_id(member_id).rank
+  end
+
+  @doc """
+  members=
+  [
+             %{rating: 8, rank: 4, member_id: 100},
+             %{rating: 5, rank: 0, member_id: 4},
+             %{rating: 6, rank: 0, member_id: 2},
+             %{rating: 17, rank: 0, member_id: 3}
+           ]
+
+  Output: Members will be sorted by rank descending; however, rank=0 players will always be last
+  """
+  def sort_members(members) do
+    non_noobs = Enum.filter(members, fn x -> x.rank != 0 end)
+    noobs = Enum.filter(members, fn x -> x.rank == 0 end)
+
+    [
+      Enum.sort_by(non_noobs, fn x -> x.rating end, &>=/2),
+      Enum.sort_by(noobs, fn x -> x.rating end, &>=/2)
+    ]
+    |> List.flatten()
+  end
+
+  @doc """
+  member_list: A sorted list of members e.g.
+  [
+  %{rating: 8, member_id: 1, rank: 0},
+  %{rating: 5, member_id: 4, rank: 0},
+  %{rating: 6, member_id: 2, rank: 0},
+  %{rating: 7, member_id: 3, rank: 0}
+  ]
+
+  Returns %{teams:teams, logs:logs}
+  """
+  def assign_teams(member_list, number_of_teams) do
+    default_acc = %{teams: create_empty_teams(number_of_teams), logs: ["Begin split_one_chevs balance"]}
+
+    Enum.reduce(member_list, default_acc, fn x, acc ->
+      picking_team = get_picking_team(acc.teams)
+      update_picking_team = Map.merge(picking_team, %{members: [x | picking_team.members]})
+      username = Account.get_username_by_id(x.member_id)
+      new_log = "#{username} (Chev: #{x.rank+1}) picked for Team #{picking_team.team_id}"
+
+      %{
+        teams: [update_picking_team | get_non_picking_teams(acc.teams, picking_team)],
+        logs: acc.logs ++ [new_log]
+      }
+    end)
+  end
+
+  @spec create_empty_teams(any()) :: any()
+  def create_empty_teams(count) do
+    for i <- 1..count,
+        do: %{team_id: i, members: []}
+  end
+
+  @spec get_picking_team(any()) :: any()
+  def get_picking_team(teams) do
+    default_picking_team = Enum.at(teams, 0)
+
+    Enum.reduce(teams, default_picking_team, fn x, acc ->
+      # Team is picker if it has least members
+      if(length(x.members) < length(acc.members)) do
+        x
+      else
+        # Team is picker if it is tied for least and has lower team rating
+        if(
+          length(x.members) == length(acc.members) && get_team_rating(x) < get_team_rating(acc)
+        ) do
+          x
+        else
+          acc
+        end
+      end
```

Comment:
I believe it's more idiomatic in elixir to use [`cond`](https://hexdocs.pm/elixir/case-cond-and-if.html) over these nested ifs.